ingestion/src/metadata/ingestion/source/dashboard/quicksight/metadata.py

```diff
@@ -161,61 +186,97 @@ def yield_dashboard_chart(
+                    ):
+                        logger.warning(
+                            "We currently don't support lineage to S3 Source or Custom Sql Queries"
+                        )
+                        continue
+
+                    # db_name = data_source
+                    schema_name = data_source["RelationalTable"]["Schema"]
+                    table_name = data_source["RelationalTable"]["Name"]
+
+                    list_data_source_func = lambda kwargs: self.client.list_data_sources(  # pylint: disable=unnecessary-lambda-assignment
+                        **kwargs
+                    )
+
+                    data_source_summary_list = self._check_pagination(
+                        listing_method=list_data_source_func,
+                        entity_key="DataSources",
+                    )
+
+                    data_source_ids = [
+                        data_source_arn["DataSourceId"]
+                        for data_source_arn in data_source_summary_list
+                        if data_source_arn["Arn"]
+                        in data_source["RelationalTable"]["DataSourceArn"]
+                    ]
+
+                    for data_source_id in data_source_ids:
+                        data_source_dict = self.client.describe_data_source(
+                            AwsAccountId=self.aws_account_id,
+                            DataSourceId=data_source_id,
+                        )["DataSource"]["DataSourceParameters"]
+                        for db in data_source_dict.keys():
+                            from_fqn = None
+                            try:
+                                database_name = data_source_dict[db]["Database"]
+                            except Exception as err:
+                                logger.error(err)
+                                logger.debug(traceback.format_exc())
+                                database_name = None
+                                from_fqn = fqn.build(
+                                    self.metadata,
+                                    entity_type=Table,
+                                    service_name=db_service_name,
+                                    table_name=table_name,
+                                    schema_name=schema_name,
+                                    database_name="",
+                                )
+                            from_entity = self.metadata.get_by_name(
+                                entity=Table,
+                                fqn=f"{db_service_name}{fqn.FQN_SEPARATOR}"
+                                f"{database_name}{fqn.FQN_SEPARATOR}{schema_name}{fqn.FQN_SEPARATOR}{table_name}"
+                                if database_name
+                                else from_fqn,
+                            )
+
+                            to_entity = self.metadata.get_by_name(
+                                entity=Dashboard,
```

Comment:
dashboard fqn can also be build from fqn builder

Reply:
imo additional es calls can be avoided if we have a full fqn available

Reply:
building dashboard fqn via the fqn builder does not make calls to es

Reply:
ohh, right..
let me add that